staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -693,6 +693,8 @@ type APIResource struct {
 	Verbs Verbs `json:"verbs" protobuf:"bytes,4,opt,name=verbs"`
 	// shortNames is a list of suggested short names of the resource.
 	ShortNames []string `json:"shortNames,omitempty" protobuf:"bytes,5,rep,name=shortNames"`
+	// groupNames is a list of group aliases this resource belongs to (e.g. 'all')
+	GroupNames []string `json:"groupNames,omitempty" protobuf:"bytes,6,rep,name=groupNames"`
```

Comment:
> That does raise some interesting name-squatting concerns. Perhaps a CLI user should be forced to opt-in to particular categories in some way?

I don't think so, at least for now. Categories become a cluster-wide concept with this, and part of the "contract" you accept by interacting with the given cluster (set of api object it exposes, aliases for them, and now categories).